### PR TITLE
Add more docstrings and type hints

### PR DESCRIPTION
**Pull request by jan-janssen**
This is the second part following https://github.com/pyiron/pyiron_base/pull/1640

### CODE REVIEW

**Comment by samwaseda** [comment]
I’m not sure if I’d import `argparse` just for the type hinting. Don’t you want to do a string type hinting?

**Comment by jan-janssen** [comment]
> I’m not sure if I’d import `argparse` just for the type hinting. Don’t you want to do a string type hinting?

In `pyiron_base.cli.control` the `argparse` package is required to parse the command line arguments. The other command line interfaces in `pyiron_base.cli.*` are only imported in `pyiron_base.cli.control`, so it does not matter if we already import `argparse` in the command line interfaces of in `pyiron_base.cli.control`, the number of modules imported when using one of the command line utilities remains the same.

File: pyiron_base/database/generic.py [hunk 1]
```diff
@@ -253,19 +255,19 @@ def view_mode(self, value):
...
-        element_lst=None,
-    ):
+        sql_query: str,
+        user: str,
+        project_path: str,
+        recursive: bool,
+        job: Optional[str] = None,
+        sub_job_name: str = "%",
+        element_lst: List[str] = None,
+    ) -> Union[List[dict]]:
```
Comment (samwaseda) [inline pyiron_base/database/generic.py]: What does this mean

**Review by samwaseda — APPROVED** [review]
Just a nit otherwise it looks good to me

File: pyiron_base/database/generic.py (hunk 1 above)
Comment (samwaseda) [reply pyiron_base/database/generic.py]: I mean `Union[List[dict]]`

Comment (jan-janssen) [reply pyiron_base/database/generic.py]: Good point - I fixed it.

Comment (samwaseda) [reply pyiron_base/database/generic.py]: Did `copilot` make this mistake?

**Pull request merged**